x-pack/plugins/endpoint/public/applications/endpoint/index.tsx

```diff
@@ -52,6 +66,101 @@ interface RouterProps {
   store: Store;
   coreStart: CoreStart;
   depsStart: EndpointPluginStartDependencies;
+  ingestManager: IngestManagerSetup;
+}
+
+const IsAppUnavailable: React.FunctionComponent<{
+  ingestManager: IngestManagerSetup;
+  setIsAppUnavailable: React.Dispatch<React.SetStateAction<boolean>>;
+}> = ({ ingestManager, setIsAppUnavailable }) => {
+  (async () => {
+    if (await isIngestManagerInitialized(ingestManager)) {
+      setIsAppUnavailable(false);
+      return null;
+    }
+  })();
+
+  const [hasIngestSetupFinished, setHasIngestSetupFinished] = React.useState(false);
+  const [isIngestInitSuccessful, setIsIngestInitSuccessful] = React.useState(false);
+  const [ingestInitError, setIngestInitError] = React.useState<Error | undefined>(undefined);
+  React.useEffect(() => {
+    ingestManager
+      .setup()
+      .then(response => {
+        setHasIngestSetupFinished(true);
+        if (response.error) {
+          setIngestInitError(response.error);
+        } else {
+          setIsIngestInitSuccessful(response.data?.isInitialized || false);
+        }
+      })
+      .catch(error => {
+        setHasIngestSetupFinished(true);
+        setIngestInitError(error);
+      });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (!hasIngestSetupFinished) {
+    return (
+      <EuiFlexGroup justifyContent="spaceAround">
+        <EuiFlexItem grow={false}>
+          <EuiLoadingSpinner size="xl" />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+
+  const errorText =
+    `Response from Ingest Manager: ${ingestInitError?.message}` ||
+    'Ingest Manager failed to initialize for an unknown reason';
+  if (!isIngestInitSuccessful || ingestInitError) {
+    return (
+      <EuiPage className="ingestErrorPage">
+        <EuiPageBody>
+          <EuiPageHeader>
+            <EuiPageHeaderSection className="ingestErrorHeader">
+              <EuiTitle size="m">
+                <h1 data-test-subj="ingestErrorTitle">
+                  <FormattedMessage
+                    id="xpack.endpoint.endpointAppTitle"
+                    defaultMessage="Endpoint App"
+                  />
+                </h1>
+              </EuiTitle>
+            </EuiPageHeaderSection>
+          </EuiPageHeader>
+          <EuiPageContent className="ingestErrorContent">
+            <EuiText>
+              <h2>
+                <FormattedMessage
+                  id="xpack.endpoint.ingestErrorHeader"
+                  defaultMessage="Ingest Manager Initialization Failure"
+                />
+              </h2>
+              <p>
+                <FormattedMessage
+                  id="xpack.endpoint.ingestErrorMessage"
+                  defaultMessage={errorText}
+                />
+              </p>
+            </EuiText>
+          </EuiPageContent>
+        </EuiPageBody>
+      </EuiPage>
+    );
+  }
+
+  setIsAppUnavailable(false);
```

Comment:
I think this is essentially triggering an update to state during `render()` - do you see any react errors in the browser console?

Might need to move it to a `useEffect()` block.